ghci-wrapper/src/Language/Haskell/GhciWrapper.hs

```diff
@@ -94,10 +102,12 @@ close repl = do
   when (e /= ExitSuccess) $ do
     throwIO (userError $ "Language.Haskell.GhciWrapper.close: Interpreter exited with an error (" ++ show e ++ ")")
 
-putExpression :: Interpreter -> String -> IO ()
-putExpression Interpreter{hIn = stdin} e = do
+putExpression :: Interpreter -> Bool -> String -> IO ()
+putExpression Interpreter{hIn = stdin} preserveIt e = do
   hPutStrLn stdin e
+  when preserveIt $ hPutStrLn stdin $ "let " ++ itSafe ++ " = it"
   hPutStrLn stdin marker
+  when preserveIt $ hPutStrLn stdin $ "let it = " ++ itSafe
```

Comment:
The `it` variable isn't defined until the first expression is evaluated which sets it, e.g.

```
% ghci
GHCi, version 8.0.2: http://www.haskell.org/ghc/  :? for help
Prelude> it

<interactive>:1:1: error:
    • Variable not in scope: it
    • Perhaps you meant ‘id’ (imported from Prelude)
Prelude> :t 'a'
'a' :: Char
Prelude> it

<interactive>:3:1: error:
    • Variable not in scope: it
    • Perhaps you meant ‘id’ (imported from Prelude)
```

---

Which makes me notice that there's a bug, if the first doctest is e.g. `:t ()`, the `doctest` will fail.

I'll try to fix that shortly.